### DISCUSSION TITLE
How do i integrate theia into my electron application

### DISCUSSION THREAD

**Discussion by chenqianwan** [eclipse-theia/theia#9157] [Q&A]
I am a newbie. I already have such an electron desktop application. I want to replace the following part with theia. What should I do? Thank you
![electron](https://user-images.githubusercontent.com/40291085/110321199-4c3f0380-804c-11eb-95d4-a7edd9bfc18c.png)

**Comment by vince-fugnitto** [comment]
@chenqianwan I do not believe there are any examples on how to do so, the framework already supports creating `electron` applications and I doubt we ever attempted to embed it into an already existing electron app:

- https://www.electronjs.org/docs/tutorial/web-embeds

You might be able to embed the app as an `iframe` however I'm not sure how well this will work for `electron`.

- https://github.com/eclipse-theia/theia/issues/1045
- https://github.com/eclipse-theia/theia/issues/3823
